packages/@orbit/record-cache/src/operators/sync-query-operators.ts

```diff
@@ -163,6 +163,8 @@ function applyFilter(record: Record, filter: any) {
             actual !== undefined &&
             expected.some(e => actual.type === e.type && actual.id === e.id)
           );
+        } else if (expected === null) {
+          return !actual;
```

Comment:
We need to be sure that an `undefined` relationship is treated as indeterminate, while `null` represents an actually empty relationship.

```suggestion
          return actual === null;
```